File: AFMobile/AFAndroid/app/src/main/java/cz/cvut/fel/matyapav/afandroid/builders/ListBuilder.java
package cz.cvut.fel.matyapav.afandroid.builders;

import android.graphics.Paint;
import android.graphics.drawable.ShapeDrawable;
import android.graphics.drawable.shapes.RectShape;
import android.view.View;
import android.widget.LinearLayout;
import android.widget.ListView;

import com.tomscz.afswinx.common.ParameterMissingException;

import org.json.JSONException;

import cz.cvut.fel.matyapav.afandroid.AFAndroid;
import cz.cvut.fel.matyapav.afandroid.components.types.AFComponent;
import cz.cvut.fel.matyapav.afandroid.components.types.AFList;
import cz.cvut.fel.matyapav.afandroid.enums.SupportedComponents;

/**
 * Builds list component from class definition
 *
 * @author Pavel Matyáš (matyapav@fel.cvut.cz).
 * @since 1.0.0..
 */
public class ListBuilder extends AFComponentBuilder<ListBuilder> {

    @Override
    public AFList createComponent() throws Exception {
        try {
            initializeConnections();
        } catch (ParameterMissingException | JSONException e) {
            e.printStackTrace();
            return null;
        }
        String modelResponse = getModelResponse();
        //create form from response
        AFList list = (AFList) buildComponent(modelResponse, SupportedComponents.LIST);
        //fill it with data (if there are some)
        String data = getDataResponse();
        if(data != null) {
            list.insertData(data);
        }
        AFAndroid.getInstance().addCreatedComponent(getComponentKeyName(), list);
        return list;
    }

    @Override
    protected View buildComponentView(AFComponent component) {
        ListView listView = new ListView(getContext());
        LinearLayout.LayoutParams params = new LinearLayout.LayoutParams(getSkin().getListWidth(), getSkin().getListHeight());
        params.setMargins(getSkin().getComponentMarginLeft(), getSkin().getComponentMarginTop(),
                getSkin().getComponentMarginRight(), getSkin().getComponentMarginBottom());
        listView.setLayoutParams(params);
        listView.setAdapter(null);
        //create border
        if(getSkin().getListBorderWidth() > 0) {
            RectShape rect = new RectShape();
            ShapeDrawable rectShapeDrawable = new ShapeDrawable(rect);
            Paint paint = rectShapeDrawable.getPaint();
            paint.setColor(getSkin().getListBorderColor());
            paint.setStyle(Paint.Style.STROKE);
            paint.setStrokeWidth(getSkin().getListBorderWidth());
            listView.setBackground(rectShapeDrawable);
        }

        listView.setScrollbarFadingEnabled(!getSkin().isListScrollBarAlwaysVisible());
        //listView.setBackgroundColor(getSkin().getListBackgroundColor());
        ((AFList)component).setListView(listView);
        return listView;
    }




}

File: AFMobile/AFWindowsPhone/AFWindowsPhone/builders/ListBuilder.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using AFWindowsPhone.builders.components.types;
using Windows.UI.Xaml;
using AFWindowsPhone.enums;
using Windows.UI.Xaml.Controls;
using Windows.UI.Xaml.Media;

namespace AFWindowsPhone.builders
{
    class ListBuilder : AFComponentBuilder<ListBuilder>
    {
        public override AFComponent createComponent()
        {
            initializeConnections();
            var modelTask = Task.Run(getModelResponse);
            modelTask.Wait();
            String modelResponse = modelTask.Result;
            //create list from response
            AFList list = (AFList)buildComponent(modelResponse, SupportedComponents.LIST);
            //fill it with data (if there are some)
            var dataTask = Task.Run(getDataResponse);
            dataTask.Wait();
            String data = dataTask.Result;
            if (data != null)
            {
                list.insertData(data);
            }
            AFWindowsPhone.getInstance().addCreatedComponent(getComponentKeyName(), list);
            return list;
        }

        protected override FrameworkElement buildComponentView(AFComponent component)
        {
            ListView listView = new ListView();
            if (getSkin().getListWidth() < 0)
            {
                listView.HorizontalAlignment = getSkin().getListHorizontalAlignment();
            }
            else {
                listView.Width = getSkin().getListWidth();
            }
            if(getSkin().getListHeight() < 0)
            {
                listView.VerticalAlignment = getSkin().getListVerticalAlignment();
            }
            else
            {
                listView.Height = getSkin().getListHeight();
            }         
      
            listView.Margin = new Thickness(getSkin().getComponentMarginLeft(), getSkin().getComponentMarginTop(),
                getSkin().getComponentMarginRight(), getSkin().getComponentMarginBottom());
            //create border
            if (getSkin().getListBorderWidth() > 0)
            {
                listView.BorderBrush = new SolidColorBrush(getSkin().getListBorderColor());
                listView.BorderThickness = new Thickness(getSkin().getListBorderWidth());
            }
           
            //TODO not working 
            //set scroll bar visibility
            ScrollViewer.SetVerticalScrollBarVisibility(listView,
                getSkin().isListScrollBarAlwaysVisible() ? ScrollBarVisibility.Visible : ScrollBarVisibility.Auto);

            //listView.setBackgroundColor(getSkin().getListBackgroundColor());
            ((AFList)component).setListView(listView);
            return listView;
        }
    }
}
